Gems/PhysX/Code/Source/System/PhysXSystem.cpp

```diff
@@ -46,6 +50,51 @@ namespace PhysX
     AZ_CVAR(bool, physx_reportTimestepWarnings, false, nullptr, AZ::ConsoleFunctorFlags::Null, "A flag providing ability to turn on/off reporting of PhysX timestep warnings");
 #endif
 
+    // A helper function.
+    AZ::Debug::PerformanceCollector::DataLogType GetDataLogTypeFromCVar(const AZ::CVarFixedString& newCaptureType)
+    {
+        if (newCaptureType[0] == 'a' || newCaptureType[0] == 'A')
+        {
+            return AZ::Debug::PerformanceCollector::DataLogType::LogAllSamples;
+        }
+        else
+        {
+            return AZ::Debug::PerformanceCollector::DataLogType::LogStatistics;
+        }
+    }
```

Comment:
`fixed_string<N>::starts_with` should be used here as this will cause an out of bound access if the `newCaptureType` is an empty string

Reply:
awesome, thanks! 
Updated the Atom code with the fix too